src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -11588,35 +11587,34 @@ class CL1
+                //         System.Action<CL1?> x1 = (CL1 p1) => p1 = M1();
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "M1()").WithLocation(12, 51),
+                // (12,34): warning CS8622: Nullability of reference types in type of parameter 'p1' of 'lambda expression' doesn't match the target delegate 'Action<CL1?>'.
+                //         System.Action<CL1?> x1 = (CL1 p1) => p1 = M1();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "(CL1 p1) => p1 = M1()").WithArguments("p1", "lambda expression", "System.Action<CL1?>").WithLocation(12, 34),
+                // (17,59): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         System.Action<CL1?> x2 = delegate (CL1 p2) { p2 = M1(); };
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "M1()").WithLocation(17, 59),
+                // (17,34): warning CS8622: Nullability of reference types in type of parameter 'p2' of 'anonymous method' doesn't match the target delegate 'Action<CL1?>'.
+                //         System.Action<CL1?> x2 = delegate (CL1 p2) { p2 = M1(); };
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "delegate (CL1 p2) { p2 = M1(); }").WithArguments("p2", "anonymous method", "System.Action<CL1?>").WithLocation(17, 34),
+                // (24,34): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         D1 x3 = (CL1 p3) => p3 = M1();
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "M1()").WithLocation(24, 34),
+                // (24,17): warning CS8622: Nullability of reference types in type of parameter 'p3' of 'lambda expression' doesn't match the target delegate 'C.D1'.
+                //         D1 x3 = (CL1 p3) => p3 = M1();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "(CL1 p3) => p3 = M1()").WithArguments("p3", "lambda expression", "C.D1").WithLocation(24, 17),
+                // (29,42): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         D1 x4 = delegate (CL1 p4) { p4 = M1(); };
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "M1()").WithLocation(29, 42),
+                // (29,17): warning CS8622: Nullability of reference types in type of parameter 'p4' of 'anonymous method' doesn't match the target delegate 'C.D1'.
+                //         D1 x4 = delegate (CL1 p4) { p4 = M1(); };
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "delegate (CL1 p4) { p4 = M1(); }").WithArguments("p4", "anonymous method", "C.D1").WithLocation(29, 17)
                 );
         }
 
-        // PROTOTYPE(NullableReferenceTypes): Calculate lamba conversion.
-        [Fact(Skip = "TODO")]
+        [Fact]
```

Comment:
I'm surprised there is a warning on x1, because of the definition of `Action<in T>`.
I would think that a method that accepts a nullable input can be converted to a delegate signature that needs non-nullable input.

Reply:
This corresponds to the conversion behavior for lambda parameters in initial binding, but we could consider relaxing this for nullability. Added a `// PROTOTYPE` comment to `NullableWalker.ReportNullabilityMismatchWithTargetDelegate` for now.

---
In reply to: [192622959](https://github.com/dotnet/roslyn/pull/27283#discussion_r192622959) [](ancestors = 192622959)